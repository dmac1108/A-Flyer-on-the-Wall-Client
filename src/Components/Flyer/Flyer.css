.list-group{
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-content: flex-start;
}

img {
    width: 25%;
    height: 25%;
}

.flyer {
    border: solid #46535E 2px;
}

dt{
    font-weight: bold;
}

img:hover {
    width: 50%;
    height: 50%;
  }

 .add-to-calendar {
    
        border-radius: 5px;
        margin-right: auto;
        margin-left: auto;
        width: 200px;
        font-size: 20px;
        color: #2A465E;
        background-color: #E2A66F ;
        text-align: center;
    
 } 

 .add-to-calendar > a:hover {
        color: #2A465E;
 }
