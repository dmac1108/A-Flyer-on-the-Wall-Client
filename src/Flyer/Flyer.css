.list-group{
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-content: flex-start;
}

img {
    width: 25%;
    height: 25%;
}