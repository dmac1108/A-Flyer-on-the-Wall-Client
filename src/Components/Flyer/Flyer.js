import config from '../../config'
import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import FlyersContext from '../../FlyersContext'
import './Flyer.css';
import AddToCalendar from 'react-add-to-calendar';
import moment from 'moment';
import FlyerApiService from '../../services/flyer-api-service'
import piexif from 'piexifjs'


 class Flyer extends Component{

    static contextType = FlyersContext

    onDeleteFlyer = (flyerId) =>{
        FlyerApiService.deleteFlyer(flyerId)
        .then(()=>{
            this.context.onDeleteFlyer(flyerId)
        })
    }

    componentDidMount(){
        const canvas = this.refs.canvas
        const ctx = canvas.getContext("2d")
        const image = this.props.image
        var exifObj = piexif.load(image)
        const srcOrientation = exifObj["0th"][piexif.ImageIFD.Orientation]
    //    if(srcOrientation !== undefined){
        const width = image.width
        const height = image.height

        if (4 < srcOrientation && srcOrientation < 9) {
            console.log('in between 4 and 9')
            canvas.width = height;

            canvas.height = width;
        } else {
            console.log('otherwise')
            canvas.width = width;
            canvas.height = height;
         }

         switch (srcOrientation) {
            case 2: ctx.transform(-1, 0, 0, 1, width, 0); break;
            case 3: ctx.transform(-1, 0, 0, -1, width, height); break;
            case 4: ctx.transform(1, 0, 0, -1, 0, height); break;
            case 5: ctx.transform(0, 1, 1, 0, 0, 0); break;
            case 6: ctx.transform(0, 1, -1, 0, height, 0); break;
            case 7: ctx.transform(0, -1, -1, 0, height, width); break;
            case 8: ctx.transform(0, -1, 1, 0, 0, width); break;
            default: break;
          }

          //console.log(image)
        
          let newImage = new Image();
          newImage.onload = function(){
            ctx.drawImage(newImage, 0,0);
          }
          newImage.src = image
        // }


    }
    
    render(){
    const {id, title, location, image, eventstartdate, eventenddate, actiondate, action, category} = this.props
    const {flyers_children, children} = this.context  
    
    let flyer_children
    if(flyers_children !== null){
        flyer_children = flyers_children.filter((flyer_child)=>
        flyer_child.flyerid === id
    )
    }
    
    
    let childrenList;

    if(flyer_children !== null){
        const childrenToList = flyer_children.map((flyer_child) => children.find(child => child.id === flyer_child.childid))
        
        childrenList = childrenToList.map((flyerchild)=> <dd key={flyerchild.id}>{flyerchild.childname}</dd>)
        
    }
    
    const startTime = moment(eventstartdate).format();
    const endTime= moment(eventenddate).format()
    const description = `${config.API_ENDPOINT}/${id}`
    const event = {
            title: title,
            location: location,
            description: description,
            startTime: startTime,
            endTime: endTime,
        }
    
     
    
    return(
        <div className="flyer">
        <h2>{title}</h2>
        <canvas ref="canvas" width={425} height={640}/>
        <img ref="image" alt="Flyer Thumbnail" src={image}/>
        <dl>
            <div className="list-group">
                <dt>Location:</dt>
                <dd>{location}</dd>
            </div>
            <div className="list-group">
                <dt>Event Start Date/Time:</dt>
                <dd>{moment(eventstartdate).format('MMMM Do, h:mm a')}</dd>
               
            </div>
            <div className="list-group">
                <dt>Event End Date/Time:</dt>
                <dd>{moment(eventenddate).format('MMMM Do, h:mm a')}</dd>
                
            </div>
            <div className="list-group">
                <dt>Action Date:</dt>
                <dd>{new Date(actiondate).toString().substring(0,10)}</dd>
            </div>
            <div className="list-group">
                <dt>Action:</dt>
                <dd>{action}</dd>
            </div>
            <div className="list-group">
                <dt>Category:</dt>
                <dd>{category}</dd>
            </div>
            
                <dt>Children:</dt>
                {childrenList}
           
        </dl>
        
        <AddToCalendar event={event} buttonWrapperClass="add-to-calendar"/>
    
        <Link to={`/edit-flyer/${id}`}><button>Edit</button></Link>
        <button onClick={()=>this.onDeleteFlyer(id)}>Delete</button>
     </div>
    )
    }
}

export default Flyer