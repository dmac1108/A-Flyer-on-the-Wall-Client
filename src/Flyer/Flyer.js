import config from '../config'
import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import FlyersContext from '../FlyersContext'
import './Flyer.css';
import AddToCalendar from 'react-add-to-calendar';
import moment from 'moment';


 class Flyer extends Component{
     
    static contextType = FlyersContext

    render(){
    const {id, title, location, image, eventstartdate, eventenddate, actiondate, action, category, childid} = this.props
      
    const {flyers_children} = this.context    
    const childListItems = !childid ? '' : childid.map((childid) =>{
        const child = flyers_children.filter(child => child.id == childid)
        return <dd key={childid}>{child[0].name}</dd>
    }) 
    
    const startTime = moment(eventstartdate).format();
    const endTime= moment(eventenddate).format()
    const description = `${config.API_ENDPOINT}/${id}`
    const event = {
            title: title,
            location: location,
            description: description,
            startTime: startTime,
            endTime: endTime,
        }
    
    return(
        <div className="flyer">
        <h2>{title}</h2>
        
        <img alt="Flyer Thumbnail" src={image}/>
        <dl>
            <div className="list-group">
                <dt>Location:</dt>
                <dd>{location}</dd>
            </div>
            <div className="list-group">
                <dt>Event Start Date/Time:</dt>
                <dd>{moment(eventstartdate).format('MMMM Do, h:mm a')}</dd>
                {/*<dd>{new Date(eventdate).toString().substring(0,10)}</dd>*/}
            </div>
            <div className="list-group">
                <dt>Event End Date/Time:</dt>
                <dd>{moment(eventenddate).format('MMMM Do, h:mm a')}</dd>
                {/*<dd>{new Date(eventdate).toString().substring(0,10)}</dd>*/}
            </div>
            <div className="list-group">
                <dt>Action Date:</dt>
                <dd>{new Date(actiondate).toString().substring(0,10)}</dd>
            </div>
            <div className="list-group">
                <dt>Action:</dt>
                <dd>{action}</dd>
            </div>
            <div className="list-group">
                <dt>Category:</dt>
                <dd>{category}</dd>
            </div>
            
                <dt>Children:</dt>
                {childListItems}
           
        </dl>
        
        <AddToCalendar event={event} buttonWrapperClass="add-to-calendar"/>
    
        <Link to={`/edit-flyer/${id}`}><button>Edit</button></Link>
        <button onClick={()=>this.context.onDeleteFlyer(id)}>Delete</button>
     </div>
    )
    }
}

export default Flyer