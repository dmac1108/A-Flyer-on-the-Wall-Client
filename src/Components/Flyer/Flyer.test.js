import React from 'react';
import ReactDOM from 'react-dom';
import Flyer from './Flyer'
import {BrowserRouter } from 'react-router-dom'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'



const testFlyer = {
    id: 1,
    title: 'Test',
    image:"require('./campingbase64.bin')",
    eventstartdate: "9/3/19 9:30",
    eventenddate: "9/4/19 10:30",
    actiondate:'8/15/19',
    action: 'Test render', 
    flyercategory: 'Scouts', 
    childid: [1,2],
    category: 1
}

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><Flyer id={testFlyer.id} title={testFlyer.title} image={testFlyer.image} eventdate={testFlyer.eventdate} actiondate={testFlyer.actiondate} action={testFlyer.action} category={testFlyer.category} childid={testFlyer.childid} /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
}); 


it('renders the UI as expected', () =>{
    const wrapper = shallow(<BrowserRouter><Flyer id={testFlyer.id} title={testFlyer.title} image={testFlyer.image} eventdate={testFlyer.eventdate} actiondate={testFlyer.actiondate} action={testFlyer.action} category={testFlyer.category} childid={testFlyer.childid}/></BrowserRouter>)
    expect(toJson(wrapper)).toMatchSnapshot()
})

