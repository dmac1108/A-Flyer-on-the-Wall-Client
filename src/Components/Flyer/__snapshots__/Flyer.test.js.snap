// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Flyer
    action="Test render"
    actiondate="8/15/19"
    category={1}
    childid={
      Array [
        1,
        2,
      ]
    }
    id={1}
    image="require('./campingbase64.bin')"
    title="Test"
  />
</Router>
`;
